deps/rabbit/src/rabbit_queue_location_min_masters.erl

```diff
@@ -6,65 +6,60 @@
-                                    <<"min-masters">>, ?MODULE]}},
-                    {requires,    rabbit_registry},
-                    {enables,     kernel_ready}]}).
+-rabbit_boot_step(
+    {?MODULE, [
+        {description, "locate queue master min bound queues"},
+        {mfa,
+            {rabbit_registry, register, [
+                queue_master_locator,
+                <<109, 105, 110, 45, 109, 97, 115, 116, 101, 114, 115>>,
+                ?MODULE
+            ]}},
+        {requires, rabbit_registry},
+        {enables, kernel_ready}
+    ]}
+).
 
 %%---------------------------------------------------------------------------
 %% Queue Master Location Callbacks
 %%---------------------------------------------------------------------------
 
 description() ->
-    [{description,
-      <<"Locate queue master node from cluster node with least bound queues">>}].
+    [
+        {description, <<
+            "Locate queue master node from cluster node with least bound "
+            "queues"
+        >>}
+    ].
 
 queue_master_location(Q) when ?is_amqqueue(Q) ->
-    Cluster = rabbit_queue_master_location_misc:all_nodes(Q),
-    QueueNames = rabbit_amqqueue:list_names(),
-    MastersPerNode0 = lists:foldl(
-        fun(#resource{virtual_host = VHost, name = QueueName}, NodeMasters) ->
-            case rabbit_queue_master_location_misc:lookup_master(QueueName, VHost) of
-                {ok, Master} when is_atom(Master) ->
-                    case maps:is_key(Master, NodeMasters) of
-                        true -> maps:update_with(Master,
-                                                 fun(N) -> N + 1 end,
-                                                 NodeMasters);
-                        false -> NodeMasters
-                    end;
-                _ -> NodeMasters
-            end
-        end,
-        maps:from_list([{N, 0} || N <- Cluster]),
-        QueueNames),
-    
-    MastersPerNode = maps:filter(fun (Node, _N) ->
-                                    not rabbit_maintenance:is_being_drained_local_read(Node)
-                                 end, MastersPerNode0),
-
-    case map_size(MastersPerNode) > 0 of
-        true ->
-            {MinNode, _NMasters} = maps:fold(
-                fun(Node, NMasters, init) ->
-                    {Node, NMasters};
-                (Node, NMasters, {MinNode, MinMasters}) ->
-                    case NMasters < MinMasters of
-                        true  -> {Node, NMasters};
-                        false -> {MinNode, MinMasters}
-                    end
-                end,
-                init, MastersPerNode),
-            {ok, MinNode};
-        false ->
-            undefined
+    Nodes0 = rabbit_nodes:all_running(),
+    Nodes = rabbit_maintenance:filter_out_drained_nodes_local_read(Nodes0),
+    case Nodes of
+        [] ->
+            {ok, node()};
+        _ ->
+            MastersPerNode0 = erpc:multicall(Nodes, ets, info, [queue_metrics, size], 10),
```

Comment:
10ms is an incredibly low timeout, please change this to 15s which is our default in most places